controllers/spaceprovisionerconfig/spaceprovisionerconfig_controller.go

```diff
@@ -58,62 +69,163 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
+		if errors.IsNotFound(err) {
+			return nil
 		}
-		reportedError = fmt.Errorf("failed to update the SpaceProvisionerConfig status: %w", err)
+		return err
 	}
 
-	return ctrl.Result{}, reportedError
-}
+	var reason string
+	resultCondition := clusterCondition
 
-func (r *Reconciler) determineReadyState(ctx context.Context, spc *toolchainv1alpha1.SpaceProvisionerConfig) (toolchainv1alpha1.Condition, error) {
-	toolchainCluster := &toolchainv1alpha1.ToolchainCluster{}
-	toolchainClusterKey := runtimeclient.ObjectKey{Name: spc.Spec.ToolchainCluster, Namespace: spc.Namespace}
-	var toolchainPresent corev1.ConditionStatus
-	toolchainPresenceReason := toolchainv1alpha1.SpaceProvisionerConfigValidReason
-	var reportedError error
-	toolchainPresenceMessage := ""
-	if err := r.Client.Get(ctx, toolchainClusterKey, toolchainCluster); err != nil {
-		if !errors.IsNotFound(err) {
-			// we need to requeue the reconciliation in this case because we cannot be sure whether the ToolchainCluster
-			// is really present in the cluster or not. If we did not do that and instead just reported the error in
-			// the status, we could eventually leave the SPC in an incorrect state once the error condition in the cluster,
-			// that prevents us from reading the ToolchainCluster, clears. I.e. we need the requeue to keep the promise
-			// of eventual consistency.
-
-			reportedError = fmt.Errorf("failed to get the referenced ToolchainCluster: %w", err)
-			toolchainPresenceMessage = reportedError.Error()
+	if clusterCondition == corev1.ConditionTrue {
+		spc.Status.ConsumedCapacity, err = collectConsumedCapacity(ctx, r.Client, spc.Spec.ToolchainCluster, spc.Namespace)
+		if err != nil {
+			updateReadyCondition(spc, corev1.ConditionUnknown, toolchainv1alpha1.SpaceProvisionerConfigFailedToDetermineCapacityReason, err.Error())
+			return err
 		}
-		toolchainPresenceReason = toolchainv1alpha1.SpaceProvisionerConfigToolchainClusterNotFoundReason
-		toolchainPresent = corev1.ConditionFalse
-	} else {
-		readyCond, found := condition.FindConditionByType(toolchainCluster.Status.Conditions, toolchainv1alpha1.ConditionReady)
-		if !found {
-			toolchainPresent = corev1.ConditionFalse
+
+		capacityCondition := r.determineCapacityReadyState(spc)
+		if capacityCondition != corev1.ConditionTrue {
+			reason = toolchainv1alpha1.SpaceProvisionerConfigInsufficientCapacityReason
 		} else {
-			toolchainPresent = readyCond.Status
+			reason = toolchainv1alpha1.SpaceProvisionerConfigValidReason
+		}
+
+		resultCondition = capacityCondition
+	} else {
+		reason = toolchainv1alpha1.SpaceProvisionerConfigToolchainClusterNotReadyReason
+	}
+
+	readyCondition := toolchainv1alpha1.Condition{
+		Type:   toolchainv1alpha1.ConditionReady,
+		Status: resultCondition,
+		Reason: reason,
+	}
+
+	spc.Status.Conditions, _ = condition.AddOrUpdateStatusConditions(spc.Status.Conditions, readyCondition)
+
+	return nil
+}
```

Comment:
proposal for the simplification - I would deal with the ToolchainCluster status first, then address the capacity (and get rid of the extra if-else statement):
```go
// if ToolchainCluster is not ready, then set SPC as not ready too
if clusterCondition != corev1.ConditionTrue {
	updateReadyCondition(spc, clusterCondition, toolchainv1alpha1.SpaceProvisionerConfigToolchainClusterNotReadyReason, "")
	return nil
}

// check the capacity and update the status accordingly
spc.Status.ConsumedCapacity, err = collectConsumedCapacity(ctx, r.Client, spc.Spec.ToolchainCluster, spc.Namespace)
if err != nil {
	updateReadyCondition(spc, corev1.ConditionUnknown, toolchainv1alpha1.SpaceProvisionerConfigFailedToDetermineCapacityReason, err.Error())
	return err
}
capacityCondition := r.determineCapacityReadyState(spc)

reason := toolchainv1alpha1.SpaceProvisionerConfigValidReason
if capacityCondition != corev1.ConditionTrue {
	reason = toolchainv1alpha1.SpaceProvisionerConfigInsufficientCapacityReason
}

updateReadyCondition(spc, capacityCondition, reason, "")
return nil
```